src/compiler/transformers/module/module.ts

```diff
@@ -653,7 +658,17 @@ namespace ts {
                 }
             }
             else {
-                addExportMemberAssignments(resultStatements, name);
+                if (!exportEquals && exportSpecifiers && hasProperty(exportSpecifiers, name.text)) {
+                    if (!bindingNameExportSpecifiers) {
+                        bindingNameExportSpecifiers = {};
+                    }
+                    const sourceFileName = currentSourceFile.fileName;

```

Comment:
In other places we generally use the node id for this kind of map, via `getOriginalNodeId`.